app/controllers/api/storage.php

```diff
@@ -739,12 +728,12 @@
 
         $queries = [new Query('bucketId', Query::TYPE_EQUAL, [$bucketId])];
 
-        if($search) {
+        if ($search) {
             $queries[] = [new Query('name', Query::TYPE_SEARCH, [$search])];
         }
 
         if (!empty($cursor)) {
```

Comment:
I remember a bug with cursor in database, we needed to skip some check with some specific permission level, because it was saying "cursor not found". Can you double-check that is solved here?

Reply:
Yes, I have made sure to check for permission level during cursor get.